src/fault_injector/fault_types/fault.py

```diff
@@ -46,3 +61,43 @@ def next_tick(self, tick: int):
             self.inject_fault(tick=tick)
         elif tick == self.configuration.end_tick:
             self.resolve_fault(tick=tick)
+
+
+class TrainMixIn:
+    """adds the functionality to get the train in which the fault should be injected"""
+
+    def get_train(
+        self, wrapper: SimulationObjectUpdatingComponent, affected_element_id: str
+    ) -> Train:
+        """returns the train in which the requested fault
+
+        :param wrapper: wrapper component
+        :type wrapper: SimulationObjectUpdatingComponent
+        :param affected_element_id: id of the train in which the fault should be injected into
+        :type affected_element_id: str
+        :return: the train with the requested id
+        :rtype: Train
+        """
+        return [
```

Comment:
Don't forget to check, if the array contains at least one element (or exactly one) 😉

Reply:
Yes good catch! I will fix this in #353